DeusEx/Classes/DXRFixup.uc

```diff
@@ -0,0 +1,154 @@
+class DXRFixup expands DXRBase;
+
+function FirstEntry()
+{
+    local Actor a;
+    local ScriptedPawn p;
+
+    Super.FirstEntry();
+    
+    switch(dxr.localURL)
+    {
+        case "06_HONGKONG_TONGBASE":
+            foreach AllActors(class'Actor', a)
+            {
+                switch(string(a.Tag))
+                {
+                    case "AlexGone":
+                    case "TracerGone":
+                    case "JaimeGone":
+                        a.Destroy();
+                        break;
+                    case "Breakintocompound":
+                    case "LumpathPissed":
+                        Trigger(a).bTriggerOnceOnly = False;
+                        break;
+                    default:
+                        break;
+                }
+            }    
+            break;
+        case "06_HONGKONG_WANCHAI_MARKET":
+            foreach AllActors(class'Actor', a)
+            {
+                switch(string(a.Tag))
+                {
+                    case "DummyKeypad01":
+                        a.Destroy();
+                        break;
+                    case "BasementKeypad":
+                    case "GateKeypad":
+                        a.bHidden=False;
+                        break;    
+                    case "Breakintocompound":
+                    case "LumpathPissed":
+                        Trigger(a).bTriggerOnceOnly = False;
+                        break;
+                    default:
+                        break;
+                }
+            }
+            
+            break;
+        default:
+            break;
+    }
+}
+
+function AnyEntry()
+{
+    Super.AnyEntry();
+    
+    doFixup();
```

Comment:
AnyEntry gets called after FirstEntry/ReEntry, so you probably don't need to call doFixup() from BOTH AnyEntry and ReEntry